### PR TITLE
add expire old attempts scheduled job

### LINKED ISSUE #2: Remove old failed login attempt entries
Continuation of https://github.com/owncloud/security/issues/38 .

IMHO, we can add a background job and occ command to remove entries that inserted before a certain time.

### PR DESCRIPTION
**Pull request by karakayasemi**
add expire old attempts scheduled job

resolves #2

### CODE REVIEW

**Comment by karakayasemi** [comment]
sorry for the mistake, I added backgroundjob section to info.xml, and this time tested in my local.

**Comment by karakayasemi** [comment]
@DeepDiver1975 conflicts are resolved, tested in my local instance. Please review again, thanks. It is out topic, but I also need help for activating scrutinizer ci.

**Comment by karakayasemi** [comment]
all addressed. please review

**Comment by karakayasemi** [comment]
@DeepDiver1975 please review one more time when you have time. Thanks.

File: lib/Db/FailedLoginAttemptMapper.php [hunk 1]
```diff
@@ -114,4 +114,15 @@ public function deleteSuspiciousAttemptsForUidIpCombination($uid, $ip) {
 			->andWhere($builder->expr()->eq('ip', $builder->createNamedParameter($ip)))
 			->execute();
 	}
+
+	/**
+	 * @param int $threshold
```
Comment (PVince81) [inline lib/Db/FailedLoginAttemptMapper.php]: please always properly comment public methods

especially missing here is what `$threshold` is and what unit it has (milliseconds ? seconds ? days ?)

**Comment by karakayasemi** [comment]
@PVince81 I added a description for related methods. Please review again. Thanks

**Comment by karakayasemi** [comment]
I could not find a way for triggering drone to build this PR in drone API documentation. @phil-davis @individual-it have you any suggestion? Thanks.

**Comment by individual-it** [comment]
@ownclouders rebase

**Comment by individual-it** [comment]
let see if the bot does the rebase here, if not you would have to rebase manually. drone was added after you forked off, I guess that is the problem

**Comment by karakayasemi** [comment]
After the drone setup, I added a new commit and it re-triggered Travis, but Drone did not create a new job. I guess rebase is not work also. Maybe, we can merge this with admin permission or we need to create new PR.

**Comment by individual-it** [comment]
@karakayasemi I've rebased this branch on master and forced pushed, drone triggers now
please make sure you update you local branch

**Comment by karakayasemi** [comment]
@DeepDiver1975 Shall we merge?

**Pull request merged**